core/state_processor.go

```diff
@@ -940,51 +912,119 @@ func (p *StateProcessor) Process(block *types.WorkObject, batch ethdb.Batch) (ty
 		return nil, nil, nil, nil, 0, 0, 0, nil, nil, fmt.Errorf("total gas used by ETXs %d is not within the range %d to %d", totalEtxGas, minimumEtxGas, maximumEtxGas)
 	}
 
-	quaiCoinbase, err := block.QuaiCoinbase()
-	if err != nil {
-		return nil, nil, nil, nil, 0, 0, 0, nil, nil, err
+	// 50% of the fees goes to the calculation  of the averageFees generated,
+	// and this is added to the block rewards
+	halfQuaiFees := new(big.Int).Div(quaiFees, common.Big2)
+	halfQiFees := new(big.Int).Div(qiFees, common.Big2)
+
+	// convert the qi fees to quai
+	halfQiFeesInQuai := misc.QiToQuai(parent, halfQiFees)
```

Comment:
Why don't we use the prime terminus for conversions?

Reply:
we will, when i have the PR for the conversions and moving conversions from zone to prime, but in main, we still do all the exchange rate stuff in zone